deploy/charts/rook-ceph-cluster/templates/_helpers.tpl

```diff
@@ -22,3 +22,16 @@ Return the appropriate apiVersion for ingress.
 {{- print "networking.k8s.io/v1" -}}
 {{- end }}
 {{- end }}
+
+{{/*
+Remove incompatible keys from cluster spec when creating an external cluster
+*/}}
+{{- define "rook-ceph-cluster.cephClusterSpec" -}}
+{{- $cephClusterSpec := .Values.cephClusterSpec -}}
+{{- if .Values.cephClusterSpec.external.enable -}}
+{{- range tuple "dashboard" "disruptionManagement" "mon" "monitoring" "network" "storage" -}}
```

Comment:
what about mgr? Should that be removed as well?

Reply:
Surprisingly, `mgr` is not mentioned in the error message, although it is definitely checked to be empty (well, partially, only checking for an empty `mgr.modules`) : https://github.com/rook/rook/blob/master/pkg/apis/ceph.rook.io/v1/cluster.go#L55

On the other hand, `storage` is the opposite: it is in the error message, but not checked to be empty. If I find the time, I'll create a separate PR to try to sort this out.

For now, adding `mgr` to the list of keys to unset.